components/dependency_injection/autowiring.rst

```diff
@@ -0,0 +1,398 @@
+not the case (like in this example), the subsystem is smart enough to automatically
+register a private service for the ``Rot13Transformer`` class and set it as first
+argument of the `twitter_client`` service. Again, it can work only if there is one
+class of the given type. If there are several classes of the same type, you must
+use an explicit service definition or register a default implementation.
+
+As you can see, the autowiring feature drastically reduces the amount of configuration
+required to define a service. No more arguments section! It also makes it easy
+to change the dependencies of the ``TwitterClient`` class: just add or remove typehinted
+arguments in the constructor and you are done. There is no need anymore to search
+and edit related service definitions.
+
+Here is a typical controller using the ``twitter_client`` service::
+
+    // src/AppBundle/Controller/DefaultController.php
+    namespace AppBundle\Controller;
+
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+    class DefaultController extends Controller
+    {
+        /**
+         * @Route("/tweet")
+         * @Method("POST")
+         */
+        public function tweetAction(Request $request)
+        {
+            $user = $request->request->get('user');
+            $key = $request->request->get('key');
+            $status = $request->request->get('status');
+
+            if (!$user || !$key || !$status) {
+                throw new BadRequestHttpException();
+            }
+
+            $this->get('twitter_client')->tweetInRot13($user, $key, $status);
+
+            return new Response('OK');
+        }
+    }
+
+You can give a try to the API with ``curl``::
+
+    curl -d "user=kevin&key=ABCD&status=Hello" http://localhost:8000/tweet
+
+It should return ``OK``.
+
+Working with Interfaces
+-----------------------
+
+You might also find yourself using abstractions instead of implementations (especially
+in grown applications) as it allows to easily replace some dependencies without
+modifying the class depending of them.
+
+To follow this best practice, constructor arguments must be typehinted with interfaces
+and not concrete classes. It allows to replace easily the current implementation
+if necessary.
+
+Let's introduce a ``Rot13TransformerInterface``::
+
+    // src/AppBundle/Rot13TransformerInterface.php
+    namespace AppBundle;
+
+    interface Rot13TransformerInterface
+    {
+        public function transform($value);
+    }
+
+Then edit ``Rot13Transformer`` to make it implementing the new interface::
+
+    // ...
+
+    class Rot13Transformer implements Rot13TransformerInterface
```

Comment:
`Rot13TransformerInterface` doesn't make sense, why not just `TransformerInterface`. From my point of view interfaces shouldn't contain implementation details (rot13 in this case)